### ISSUE TITLE
RouterHelper : Why configure state in run() method and not config() ?

### ISSUE THREAD

**Issue by rllola** [johnpapa/generator-hottowel#74]
Can someone explain me the advantage of defining the states in the run() method instead of the config() one ?

**Comment by johnpapa** [comment]
It's a long an involved story :) I'l keep it short.

So .. I want to be able to break features into modules. Each module should be aware of its routes and set them up. Then a module responsible for layout should be able to read the routes/states along with other custom properties associated with them for menu navigation. For example, to set up a sidebar or to bar menu with awareness of where to put them, which to show, how they relate, and what order. The route and state providers do not expose the routes/states at runtime, and there was no way at config time to know when they were done collecting all of the modules' routes/states ... so I extended the provider to allow it to make it easier to collect the states and then expose a `getStates` function. I do this by allowing the $stateProvider to persist beyond the config phase.

There are other ways which I have explored too, that also work, and allow you to set routes at config time. But this way also allowed me to set routes up as the app runs, effectively allowing me to go get some routes and add them more easily.

It's not the only solution, but it does work.
